teps/0044-data-locality-and-pod-overhead-in-pipelines.md

```diff
@@ -337,24 +449,20 @@ Possible tweaks:
 
 Pros:
 * Doesn't require any changes for Pipeline or Task authors
+* Allows execution related concerns to be determined at runtime
 
 Cons:
 * Will need to update our entrypoint logic to allow for steps running in parallel
 * Doesn't give as much flexibility as being explicit
   * This functionality might not even be desirable for folks who want to make use of multiple nodes
     * We could mitigate this by adding more configuration, e.g. opt in or out at a Pipeline level, but could get
       complicated if people want more control (e.g. opting in for one workspace but not another)
+* Removes ability to run Tasks on separate pods if data is shared between them.
```

Comment:
If the reason for separate pods is security, using different service accounts would force separate pods anyways.
If the reason for separate pods is resources, having high resource requirements could enforce sequential execution of the use of separate Pods.
Alternatively this could be overcome by some kind of "anti-affinity" between `PipelineTasks`.
This gets scarily close to the scheduling domain though - which means there may be more complexity behind this than it appears on the surface.

Reply:
That's a great point. I think the reason is resources, not security, and so choosing this option would essentially be trading one resource-related problem for another.